packages/salesforcedx-vscode-core/test/vscode-integration/commands/functions/functionService.test.ts

```diff
@@ -327,4 +338,130 @@ describe('Function Service', () => {
       expect(fsSyncStub.callCount).to.equal(2);
     });
   });
+
+  describe('Debug Configuration', () => {
+    let fsSyncStub: SinonStub;
+
+    beforeEach(() => {
+      fsSyncStub = sandbox.stub(fs, 'existsSync');
+    });
+
+    it('Should set remoteRoot to /workspace when JavaScript and running in a container.', () => {
+      const functionDef: FunctionExecution = {
+        rootDir: 'FirstFunction',
+        debugPort: 7777,
+        port: 8080,
+        debugType: 'unknown',
+        terminate: () => Promise.resolve(),
+        isContainerLess: false
+      };
+
+      fsSyncStub.onCall(0).returns(false);
+      fsSyncStub.onCall(1).returns(true);
+      const service = FunctionService.instance;
+      service.registerStartedFunction(functionDef);
+      const rootDir = 'FirstFunction';
+      const functionExecution = service.getStartedFunction(rootDir);
+      const debugConfiguration = service.getDebugConfiguration(functionExecution!, rootDir);
+      expect(debugConfiguration.remoteRoot).to.equal('/workspace');
+    });
+
+    it('Should set remoteRoot to undefined when JavaScript and containerless.', () => {
+      const functionDef: FunctionExecution = {
+        rootDir: 'FirstFunction',
+        debugPort: 7777,
+        port: 8080,
+        debugType: 'unknown',
+        terminate: () => Promise.resolve(),
+        isContainerLess: true
+      };
+
+      fsSyncStub.onCall(0).returns(false);
+      fsSyncStub.onCall(1).returns(true);
+      const service = FunctionService.instance;
+      service.registerStartedFunction(functionDef);
+      const rootDir = 'FirstFunction';
+      const functionExecution = service.getStartedFunction(rootDir);
+      const debugConfiguration = service.getDebugConfiguration(functionExecution!, rootDir);
+      expect(debugConfiguration.remoteRoot).to.equal(undefined);
+    });
+
+    it('Should set remoteRoot to /workspace when TypeScript and running in a container.', () => {
+      const functionDef: FunctionExecution = {
+        rootDir: 'FirstFunction',
+        debugPort: 7777,
+        port: 8080,
+        debugType: 'unknown',
+        terminate: () => Promise.resolve(),
+        isContainerLess: false
+      };
+
+      fsSyncStub.returns(true);
+      const service = FunctionService.instance;
+      service.registerStartedFunction(functionDef);
+      const rootDir = 'FirstFunction';
+      const functionExecution = service.getStartedFunction(rootDir);
+      const debugConfiguration = service.getDebugConfiguration(functionExecution!, rootDir);
+      expect(debugConfiguration.remoteRoot).to.equal('/workspace');
+    });
+
+    it('Should set remoteRoot to /workspace when TypeScript and containerless.', () => {
```

Comment:
This might be a good reason to remove the remoteRoot on local debugging.  This test  and the next one tells us we expect this setting for locally running TS/java functions, but we don't really need it so it could lead to future confusion.

Reply:
I updated the tests (the containerless ones, for all 3 languages) to check for the prop to not be defined, and I also updated the titles of the tests.